src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -413,37 +413,10 @@ public static RevocationMode GetRevocationMode()
         /// <summary>
         /// Get a list of all the paths of the settings files used as part of this settings object
         /// </summary>
+        [Obsolete("GetConfigFilePaths(ISettings) has been deprecated. Please use ISettings.GetConfigFilePaths() instead.", error: true)]
         public static IEnumerable<string> GetConfigFilePaths(ISettings settings)
         {
-            if (settings == null)
-            {
-                throw new ArgumentNullException(nameof(settings));
-            }
-
-            if (settings is Settings settingsImpl)
-            {
-                return settingsImpl.Priority.Select(config => Path.GetFullPath(Path.Combine(config.DirectoryPath, config.FileName)));
-            }
-
-            return Enumerable.Empty<string>();
-        }
-
-        /// <summary>
-        /// Get a list of all the roots of the settings files used as part of this settings object
-        /// </summary>
-        public static IEnumerable<string> GetConfigRoots(ISettings settings)
-        {
-            if (settings == null)
-            {
-                throw new ArgumentNullException(nameof(settings));
-            }
-
-            if (settings is Settings settingsImpl)
-            {
-                return settingsImpl.Priority.Select(config => config.DirectoryPath);
-            }
-
-            return Enumerable.Empty<string>();
+            throw new NotSupportedException();
```

Comment:
We could add `VSSettings` here but that is just hacking this implementation so it works with our current implementations of `ISettings`, therefore any other implementation will always return an empty list. I'm fine with doing this for 15.9 and then remove it in 16.0 the same way we did with the other APIs, but we should also add these methods to `ISettings`

Reply:
Better yet, if we add it to `ISettings` we could make this implementation the following:
```
[Obsolete("GetConfigFilePaths(ISettings) has been deprecated. Please use ISettings.GetConfigFilePaths() instead.")]
public static IEnumerable<string> GetConfigFilePaths(ISettings settings)
{
    if (settings == null)
    {
        throw new ArgumentNullException(nameof(settings));
    }

    return settings.GetConfigFilePaths();
}
```

Therefore it will be deprecated but it will still work and we can remove it in dev16, just like the other APIs from `ISettings` and `SettingsUtility`.